build-and-test.ps1

```diff
@@ -0,0 +1,81 @@
+[cmdletbinding()]
+param(
+    [string]$VersionFilter,
+    [string]$OSFilter,
+    [switch]$IsDryRun
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+
+$buildFilter = "*"
+if (-not [string]::IsNullOrEmpty($VersionFilter))
+{
+    $buildFilter = "$VersionFilter-$buildFilter"
+}
+if (-not [string]::IsNullOrEmpty($OSFilter))
+{
+    if ([string]::IsNullOrEmpty($VersionFilter))
+    {
+        $buildFilter = "$buildFilter-$OSFilter/*"
+    }
+    else
+    {
+        $buildFilter = "$buildFilter$OSFilter/*"
+    }
+}
+
+function Get-ImageBuilder()
+{
+    $imageBuilderImageName = 'microsoft/dotnet-buildtools-prereqs:image-builder-nanoserver-20180103080524'
+    $imageBuilderContainerName = "ImageBuilder-$(Get-Date -Format yyyyMMddhhmmss)"
+
+    $attempt = 0
+    $maxRetries = 5
+    $waitFactor = 6
+    while ($attempt -lt $maxRetries)
+    {
+        try {
+            New-Item -Path "$imageBuilderFolder" -ItemType Directory -Force | Out-Null
+
+            Invoke-Expression "docker pull $imageBuilderImageName"
+            Invoke-Expression "docker create --name $imageBuilderContainerName $imageBuilderImageName"
+            Invoke-Expression "docker cp ${imageBuilderContainerName}:/image-builder $imageBuilderFolder"
+            Invoke-Expression "docker rmi -f $imageBuilderImageName"
+            break
+        }
+        catch {
+            Write-Warning "$_"
+        }
+
+        $attempt++
+        if ($attempt -ne $maxRetries)
+        {
+            $waitTime = $attempt * $waitFactor
+            Write-Host "Retry ${attempt}/${maxRetries} failed, retrying in $waitTime seconds..."
+            Start-Sleep -Seconds ($waitTime)
+            Remove-Item -Path "$imageBuilderFolder" -Recurse -Force -ErrorAction Continue
+        }
+        else {
+            throw "Retry ${attempt}/${maxRetries} failed, no more retries left."
+        }
+    }
+}
+
+$imageBuilderFolder = [System.IO.Path]::Combine("$PSScriptRoot", ".Microsoft.DotNet.ImageBuilder")
+$imageBuilderExe = [System.IO.Path]::Combine($imageBuilderFolder, "image-builder", "Microsoft.DotNet.ImageBuilder.exe")
+if (-not (Test-Path -Path "$imageBuilderExe" -PathType Leaf))
+{
+    Get-ImageBuilder
+}
+
+$imageBuilderArgs = 'build --path "$buildFilter"'
```

Comment:
Now that the tests have been checked in, you will need to pass the OSFilter and VersionFilter as build-args.

Reply:
Clarification - did you mean invoke tests as in -
`./tests/run-test.ps1 -VersionFilter $VersionFilter -OSFilter $OSFilter`

Reply:
I mean you will need to pass in `--test-var VersionFilter=$VersionFilter --test-var OSFilter=$OSFilter' into ImageBuilder

Reply:
Ok. I didn't notice the recent commit (e618fa116d493746b908b557cfbd7d6ea583c8c2) that introduced `testCommands` section in the manifest.  Yes, I will pass those args.